lib/src/model/library.dart

```diff
@@ -598,36 +597,24 @@ class Library extends ModelElement with Categorization, TopLevelContainer {
 
   HashMap<Element, Set<ModelElement>> get modelElementsMap {
     if (_modelElementsMap == null) {
-      var results = quiver.concat(<Iterable<ModelElement>>[
-        library.constants,
-        library.functions,
-        library.properties,
-        library.typedefs,
-        library.extensions.expand((e) {
-          return quiver.concat([
-            [e],
-            e.allModelElements
-          ]);
+      final results = [
```

Comment:
This seems less likely to result in a measurable performance dropoff due to caching, so if you do detect a dropoff, you might experiment with preserving this change but not others.